homeassistant/components/ezviz/camera.py

```diff
@@ -0,0 +1,276 @@
+"""This component provides basic support for Ezviz IP cameras."""
+import asyncio
+import logging
+
+from haffmpeg.tools import IMAGE_JPEG, ImageFrame
+from pyezviz.client import EzvizClient, PyEzvizError
+import voluptuous as vol
+
+from homeassistant.components.camera import PLATFORM_SCHEMA, SUPPORT_STREAM, Camera
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers import config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_CAMERAS = "cameras"
+CONF_SERIAL = "serial"
+CONF_DEFAULT_CAMERA_USERNAME = "admin"
+
+DATA_FFMPEG = "ffmpeg"
+
+CAMERAS_CONFIG = vol.Schema(
+    {
+        vol.Optional(CONF_USERNAME, default=CONF_DEFAULT_CAMERA_USERNAME): cv.string,
+        vol.Optional(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_SERIAL): cv.string,
+    }
+)
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_CAMERAS, default={}): vol.All(),
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+    }
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Ezviz IP Cameras."""
+    conf_cameras = {}
+    if CONF_CAMERAS in config:
+        conf_cameras = config[CONF_CAMERAS]
+        _LOGGER.debug("Expecting %s cameras from config", len(conf_cameras))
+
+    account = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+
+    try:
+        ezviz_client = EzvizClient(account, password)
+        ezviz_client.login()
+
+        # Get cameras
+        connections = ezviz_client.get_CONNECTION()
+        devices = ezviz_client.get_DEVICE()
+        # now, let's build the HASS devices
+        cameras = {}
+        for device in devices:
+
+            device_serial = device["deviceSerial"]
+
+            # There seem to be a bug related to localRtspPort in Ezviz API...
+            local_rtsp_port = 554
+            if connections[device_serial]["localRtspPort"] != 0:
+                local_rtsp_port = connections[device_serial]["localRtspPort"]
+
+            cameras[device_serial] = {
+                "serial": device_serial,
+                "name": device["name"],
+                "deviceType": device["deviceType"],
```

Comment:
Please use lowercase snakecase names if something else is not required by the device api.